torch/distributed/optimizer/dist_optimizer.py

```diff
@@ -0,0 +1,126 @@
+import torch.distributed.rpc as rpc
+import torch.distributed.autograd as dist_autograd
+
+from collections import defaultdict
+
+class FunctionalOptimizer:
```

Comment:
Would I be correct if I assume there is no easy way to reuse existing optimizers without modifying them as they directly reads from `param.grad`?

Reply:
I am thinking of writing an adapter that allows to use existing optim.Optimizer. In order to avoid race conditions I'll have to create multiple instance of optim.Optimizer taking the same parameters but unsharing the gradients. I can open an issue explaining the idea in more details.